src/Analysis/Ast/Impl/Caching/CacheService.cs

```diff
@@ -0,0 +1,46 @@
+﻿// Copyright(c) Microsoft Corporation
+// All rights reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the License); you may not use
+// this file except in compliance with the License. You may obtain a copy of the
+// License at http://www.apache.org/licenses/LICENSE-2.0
+//
+// THIS CODE IS PROVIDED ON AN  *AS IS* BASIS, WITHOUT WARRANTIES OR CONDITIONS
+// OF ANY KIND, EITHER EXPRESS OR IMPLIED, INCLUDING WITHOUT LIMITATION ANY
+// IMPLIED WARRANTIES OR CONDITIONS OF TITLE, FITNESS FOR A PARTICULAR PURPOSE,
+// MERCHANTABILITY OR NON-INFRINGEMENT.
+//
+// See the Apache Version 2.0 License for specific language governing
+// permissions and limitations under the License.
+
+using System.Diagnostics;
+using Microsoft.Python.Core;
+using Microsoft.Python.Core.IO;
+using Microsoft.Python.Core.Logging;
+using Microsoft.Python.Core.Services;
+
+namespace Microsoft.Python.Analysis.Caching {
```

Comment:
Thinking aloud, but I actually think this may be better served in another assembly. The `Caching` assembly is really all the "caching" that interacts with LiteDB, but not all caching in general. Perhaps all of the cache folder stuff could be moved away (a future PR, maybe).

Reply:
are you thinking putting caching stuff in Python.Core?

Reply:
Maybe just into Analysis to start. The best would be my previous suggestion of having it on IFileSystem, but the API might get tricky when it comes to overrides. Not a priority.